test/acceptance.test.ts

```diff
@@ -0,0 +1,78 @@
+import * as child from "child_process";
+import { promisify } from "util";
+
+const execPromised = promisify(child.exec);
+async function exec({
+  command,
+  args = [],
+  options = { cwd: "test" },
+}: Command): Promise<Output> {
+  const fullCommand = `${command} ${args?.join(" ")}`;
+  const execution = execPromised(fullCommand, options);
+  execution.then((ps) => {
+    console.log(fullCommand);
+    if (ps.stdout) console.log(ps.stdout);
+    if (ps.stderr) console.log(ps.stderr);
+  });
+  execution.catch(console.error);
+  return execution;
+}
+
+describe("given a git repository with a merged pr", () => {
+  beforeEach(async () => {
+    await exec({ command: "./setup.sh" });
+
+    // print and check the history graph
+    const { stdout } = await exec({
+      command: "git log --graph --oneline --decorate",
+      options: {
+        cwd: "test/repo",
+      },
+    });
+    expect(stdout).toContain(
+      "(HEAD -> master, release-2) Merge branches 'feature-b' and 'feature-c'"
+    );
+  });
+
+  describe("when backport.sh script is executed", () => {
+    beforeEach(async () => {
+      await exec({
+        command: "./backport.sh",
+        args: [
```

Comment:
Suggestion: 
You could have an object to capture the args of the script call, and then decompose the object. This would eliminate the need for the comments, give you some type safety. We can reuse this object later for other tests, and we can even create a typescript method (either here or even better in the actual production code) to call the backport script with these parameters. Then, the fact that this method is using a shell script becomes an implementation detail and you have a cleaner look in the tests.

Reply:
This is a good suggestion and we actually already have a production code module for this (see `exec.ts`). 

However, it uses `@actions/exec` to be able to run shell commands inside of the github action framework, while the acceptance test runs shell commands using its own (for ease of use) `exec` function which wraps `exec` from `child_process` (which is the default way to run shell scripts in ts). Sadly, neither can use the other and both of these have their own interfaces, so they don't mingle nicely, as far as I can tell.